pytest_html/plugin.py

```diff
@@ -157,24 +158,24 @@ def create_asset(self, content, extra_index,
         def append_extra_html(self, extra, extra_index, test_index):
             href = None
             if extra.get('format') == extras.FORMAT_IMAGE:
-                if self.self_contained:
+                content = extra.get('content')
+                if re.match('^file|^http', content):
```

Comment:
I mean if I passed `/absolute/path/to/file`, `C:\absolute\path\to\file`, or `../relative/path/to/file` then this would not work. Perhaps simply using `os.path` to check if the content is a valid path, or we could leave this as a future enhancement.